### PR TITLE
Don't show Android edit dialog when tapping read-only field

### LINKED ISSUE #7170: Text input shown on read-only fields on android
##### Issue type
 - Bug report
 
##### Minetest version
Latest commit

##### OS / Hardware

Operating system: android
CPU: god knows, Samsung S7E

##### Summary
Tapping the test in the pause dialog shows the android edit dialog. You can't change the text, however

##### Steps to reproduce
Tap the pause dialog

### PR DESCRIPTION
**Pull request by srifqi**
https://github.com/minetest/minetest/blob/a1598e1b83e28503745165d5749cd21b77e97818/doc/lua_api.txt#L2205

However, Android edit dialog is still shown when the read-only text area is tapped. This PR fixes that.

Fixes #7170.

### CODE REVIEW

File: src/gui/guiFormSpecMenu.cpp [hunk 1]
```diff
@@ -3028,29 +3028,31 @@ bool GUIFormSpecMenu::preprocessEvent(const SEvent& event)
...
 			bool retval = hovered->OnEvent(event);
-			if (retval) {
+			if (retval)
 				Environment->setFocus(hovered);
-			}
-			m_JavaDialogFieldName = getNameByID(hovered->getID());
+
+			std::string field_name = getNameByID(hovered->getID());
+			/* read-only field */
+			if (field_name == "")
```
Comment (nerzhul) [inline src/gui/guiFormSpecMenu.cpp]: please change this by field_name.empty() it's more performant

Comment (srifqi) [reply src/gui/guiFormSpecMenu.cpp]: Change below also?

Comment (nerzhul) [reply src/gui/guiFormSpecMenu.cpp]: yeah too that can be nice

Comment (srifqi) [reply src/gui/guiFormSpecMenu.cpp]: Done.

**Comment by srifqi** [comment]
Thanks for the merge!

**Pull request merged**